MyBus/MyBus/ViewController.swift

```diff
@@ -92,45 +93,65 @@ class ViewController: UIViewController, MGLMapViewDelegate, UITableViewDelegate
-        CLGeocoder().reverseGeocodeLocation(CLLocation(latitude: tappedLocation.latitude, longitude: tappedLocation.longitude))
+        CLGeocoder().reverseGeocodeLocation(CLLocation(latitude: self.destination!.latitude, longitude: self.destination!.longitude))
         {
             placemarks, error in
             if let placemark = placemarks?.first
             {
-                print()
-                let address = "\(placemark.thoroughfare! as String) \(placemark.subThoroughfare! as String)"
-
-                // Declare the marker point and set its coordinates
-                let mapPoint = MGLPointAnnotation()
-                mapPoint.coordinate = CLLocationCoordinate2D(latitude: tappedLocation.latitude, longitude: tappedLocation.longitude)
-                mapPoint.title = "Marcador"
-                mapPoint.subtitle = address
-
-                // Add marker to the map
-                self.mapView.addAnnotation(mapPoint)
-
-                // Pop-up the callout view
-                self.mapView.selectAnnotation(mapPoint, animated: true)
+                //sets attributes of annotation 
+                let annotation = MGLPointAnnotation()
+                annotation.coordinate = CLLocationCoordinate2D(latitude: self.destination!.latitude, longitude: self.destination!.longitude)
+                annotation.title = "Destino"
+                annotation.subtitle = "\(placemark.thoroughfare! as String) \(placemark.subThoroughfare! as String)"
+                //add annotation in the map
+                self.mapView.addAnnotation(annotation)
+                self.mapView.selectAnnotation(annotation, animated: true)
             }
         }
         progressNotification.stopLoadingNotification(self.view)
     }
 
+    
+    func mapView(mapView: MGLMapView, annotationCanShowCallout annotation: MGLAnnotation) -> Bool {
+        return true
+    }
+    
+    /**
+        This method sets the button of the annotation
+    */
+    func mapView(mapView: MGLMapView, rightCalloutAccessoryViewForAnnotation annotation: MGLAnnotation) -> UIView? {
+        let button = UIButton(type: .DetailDisclosure)
+        button.setImage(UIImage(named: "tabbar_route_fill"), forState: UIControlState.Normal)
+        return button
+    }
+    
+    /**
+        This method makes the search when the button is pressed on the annotation
+    */
+    func mapView(mapView: MGLMapView, annotation: MGLAnnotation, calloutAccessoryControlTapped control: UIControl) {
+        // Hide the callout view.
+        mapView.deselectAnnotation(annotation, animated: false)
+        progressNotification.showLoadingNotification(self.view)
+        //Make the search
+        let locationServiceAuth = CLLocationManager.authorizationStatus()
+        if(locationServiceAuth == .AuthorizedAlways || locationServiceAuth == .AuthorizedWhenInUse){
+           
+            let originAddress = self.mapView.userLocation?.coordinate
```

Comment:
`self.mapView.userLocation?` shouldn't be optional, we need to know or ask for a valid location to be used as origin